src/Engine/ProtoCore/Utils/CompilerUtils.cs

```diff
@@ -283,12 +284,20 @@ public static bool PreCompileCodeBlock(Core core, ref ParseParam parseParams, ID
 
             // Catch the syntax errors and errors for unsupported 
             // language constructs thrown by compile expression
+            var warnings = Check(astNodes);
+            parseParams.AppendWarnings(warnings);
+
             if (core.BuildStatus.ErrorCount > 0)
             {
                 parseParams.AppendErrors(core.BuildStatus.Errors);
                 parseParams.AppendWarnings(core.BuildStatus.Warnings);
+            }
+
+            if (parseParams.Errors.Count() > 0 || parseParams.Warnings.Count() > 0)
```

Comment:
earlier we were returning `false` only if `errors > 0`

Reply:
We should terminate earlier for variable redefinition and recursive reference. I'll update it.